instrumentation/spring/spring-integration-4.1/javaagent/src/test/groovy/SpringIntegrationAndRabbitTest.groovy

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+
+class SpringIntegrationAndRabbitTest extends AgentInstrumentationSpecification implements WithRabbitProducerConsumerTrait {
+  def setupSpec() {
+    startRabbit()
+  }
+
+  def cleanupSpec() {
+    stopRabbit()
+  }
+
+  def "should cooperate with existing RabbitMQ instrumentation"() {
+    when:
+    producerContext.getBean("producer", Runnable).run()
+
+    then:
+    assertTraces(2) {
+      trace(0, 7) {
+        span(0) {
+          name "producer"
+        }
+        span(1) {
+          name "testProducer.output"
+          childOf span(0)
+        }
+        span(2) {
+          name "exchange.declare"
+          childOf span(1)
+          kind CLIENT
+        }
+        span(3) {
+          name "testTopic -> testTopic send"
+          childOf span(1)
+          kind PRODUCER
+        }
```

Comment:
can you add attribute verification? I think that will help me understand the span structure a bit better